src/codegen.cpp

```diff
@@ -8916,6 +8916,8 @@ static jl_llvm_functions_t
             Type *RT = Arg->getParamStructRetType();
             TypeSize sz = DL.getTypeAllocSize(RT);
             Align al = DL.getPrefTypeAlign(RT);
+            if (al > MAX_ALIGN)
+                al = Align(MAX_ALIGN);
```

Comment:
Could we change these to use `dereferenceable_size` and `julia_alignment` functions? I usually distrust all DataLayout-based values, since they based on conversions to llvm types which require accuracy in first-class aggregates in matching C, which isn't entirely accurate

Reply:
I’ll give it a shot

Reply:
There doesn't seem to be a super precise type to get for all arguments sadly.